laborator/content/operatii-memorie-gdb/solution/2-delete-first/delete-first.c

```diff
@@ -0,0 +1,39 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <assert.h>
+
+char *delete_first(char *s, char *pattern)
+{
+	char *found = strstr(s, pattern);
+
+	// tratăm cazul în care pattern-ul nu apare în șirul de caractere
+	if (!found)
+		return strdup(s);
+
+	// numărul de caractere înainte de prima apariție a pattern-ului
+	int nbefore = found - s;
+	// numărul de caractere ce vor fi eliminate
+	int nremoved = strlen(pattern);
+	// alocăm exact cât avem nevoie
+	char *result = malloc(strlen(s) + 1 - nremoved);
+
+	// verificăm dacă alocarea a funcționat
+	assert(result != NULL);
+
+	// construim rezultatul
+	strncpy(result, s, nbefore);
+	strcpy(result + nbefore, found + nremoved);
+
+	return result;
+}
+
+int main(void)
+{
+	char *s = "Ana are mere";
```

Comment:
```suggestion
	char s[] = "Ana are mere";
```
Without this, the program seg-faults. There's a trick about this declaration in the skeleton [1]. Add an explanation above.

[1] https://github.com/systems-cs-pub-ro/iocla/blob/6d9ca5952005898d05d7b2bf1391606c99636a32/laborator/content/operatii-memorie-gdb/2-delete-first/delete-first.c#L9-L12

Reply:
It works because of heap allocation in the function, but you're right. Students won't necessarily implement it this way. I'll change it.